src/injectors-cache.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Contains functions that inject dynamic content.
+ */
+
+import * as brotli from './brotli-wasm-wrapper';
+import {
+  ContentEncoding,
+  HeaderKeys,
+  IncomingCloudflarePropertiesExtended,
+  supportsBrotli,
+} from './headers';
+
+const INJECTORS_CACHE_NAME = 'injectors';
+const textEncoder = new TextEncoder();
+
+let injectorsCache_: Cache;
+/** */
+async function openInjectorsCache(): Promise<Cache> {
+  if (!injectorsCache_) {
+    injectorsCache_ = await caches.open(INJECTORS_CACHE_NAME);
+  }
+  return injectorsCache_;
+}
+
+/**
+ * Generates a complete cache key for a cache object.
+ *
+ * @param parts - unique list of identifiers for object in cache.
+ * @returns a unique cache key.
+ */
+function cacheKey(...parts: string[]): string {
+  return parts.join(';');
+}
+
+/**
+ * Gets an already injected Response for the given URL.
+ *
+ * @param url - base file URL that has dynamic content injected into it.
+ * @param key - unique identifiers for this specific injection.
```

Comment:
Do we really need a double cache key (`url` and `key`)?

Reply:
I think it helps organize the cache a bit more - it splits the final cache key to the source (`url`) and to a unique identifier for each injection (which would be different between amp-geo and v0.js)